upload-agent/test.py

```diff
@@ -1,32 +1,246 @@
+        return hashlib.md5(all_file).digest()
+
+
+def get_filepath_in_test_data_dir(filename: str) -> str:
+    return os.path.abspath("./" + TEST_DATA_DIR + "/" + filename)
+
+
+def get_filepath_in_directory_for_testing(filename: str) -> str:
+    return os.path.abspath("./" + TEST_DIR + "/" + filename)
+
+
+def get_scipt_name() -> str:
+    return os.path.abspath(SCRIPT_TO_RUN)
+
+
+def run_script_in_test_dir(upload_last: bool = True):
+    args = [
+        "python",
+        get_scipt_name(),
+        "./",
+        "localhost",
+        "8080",
+    ]
+    if upload_last:
+        args.append("--upload-last")
 
-    # определяем метод `do_POST` 
+    return subprocess.Popen(args, cwd=TEST_DIR)
+
+
+class MockHTTPRequestHandler(BaseHTTPRequestHandler):
+    recieved_data = set()
+
+    # определяем метод `do_POST`
     def do_POST(self):
-        content_length = int(self.headers['Content-Length'])
-        filename = self.headers['X-File-Name']
+        content_length = int(self.headers["Content-Length"])
+        filename = self.headers["X-File-Name"]
         body = self.rfile.read(content_length)
         self.send_response(200)
-        self.send_header('Content-type', 'text/html')
+        self.send_header("Content-type", "text/html")
         self.end_headers()
         response = BytesIO()
-        response.write(filename.encode('utf-8'))
-        response.write(b':')
-        response.write(str(len(body)).encode('utf-8'))
+        response.write(filename.encode("utf-8"))
+        response.write(b":")
+        response.write(str(len(body)).encode("utf-8"))
         self.wfile.write(response.getvalue())
 
+        MockHTTPRequestHandler.recieved_data.add(hashlib.md5(body).digest())
+
+    def get_received_data():
+        return MockHTTPRequestHandler.recieved_data.copy()
+
+    def reset_received_data():
+        MockHTTPRequestHandler.recieved_data = set()
+
+
+server_address = ("", 8080)
+httpd = HTTPServer(server_address, MockHTTPRequestHandler)
+
+
+def start_server():
+    httpd.serve_forever()
+
+
+class TestUploadAgent(unittest.TestCase):
+
+    def setUp(self):
+        os.mkdir(TEST_DIR)
+        self.thread = threading.Thread(target=start_server)
+        self.thread.start()
+        return super().setUp()
+
+    def test_upload(self):
+        shutil.copyfile(
+            get_filepath_in_test_data_dir(TEST_FILE_1_CORRECT),
+            get_filepath_in_directory_for_testing(TEST_FILE_1_CORRECT),
+        )
+
+        p = run_script_in_test_dir(upload_last=True)
+
+        sleep(STANDART_TIMEOUT)
+
+        self.assertTrue(
+            get_file_hash(get_filepath_in_test_data_dir(TEST_FILE_1_CORRECT))
+            in MockHTTPRequestHandler.get_received_data()
+        )
+
+        MockHTTPRequestHandler.reset_received_data()
+
+        shutil.copyfile(
+            get_filepath_in_test_data_dir(TEST_FILE_2_CORRECT),
+            get_filepath_in_directory_for_testing(TEST_FILE_2_CORRECT),
+        )
```

Comment:
Это довольно часто встречается, вынеси в функцию
```suggestion
     def copy_test_file(filename):
        shutil.copyfile(
            get_filepath_in_test_data_dir(filename),
            get_filepath_in_directory_for_testing(filename),
        )
```